### PR TITLE
Add/extend battle flags to enable item/skill usage while interacting with NPC

### PR DESCRIPTION
**Pull request by Kenpachi2k13**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

To enable consuming usable items and using skills while interacting with a NPC I did the following:
 * Extend `item_enabled_npc` battle flag with option for usable items.
 * Add `skill_enabled_npc` battle flag.

**Issues addressed:** #862

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

File: src/map/clif.c [hunk 1]
```diff
@@ -12832,7 +12832,12 @@ static void clif_useSkillToIdReal(int fd, struct map_session_data *sd, int skill
 	// Whether skill fails or not is irrelevant, the char ain't idle. [Skotlex]
 	pc->update_idle_time(sd, BCIDLE_USESKILLTOID);
 
-	if (sd->npc_id || sd->state.workinprogress & 1) {
+#ifdef RENEWAL
+	if (sd->npc_id != 0 || (sd->state.workinprogress & 1) != 0) {
+#else
+	if ((sd->npc_id != 0 && ((tmp & INF_SELF_SKILL) == 0 || (skill->get_nk(skill_id) & NK_NO_DAMAGE) == 0))
```
Comment (4144) [inline src/map/clif.c:12838]: what if skill remove some item from inventory? or add new item?

Comment (Kenpachi2k13) [reply src/map/clif.c:12838]: Ask Gravity? :grin:

I don't know. There may be some rare cases where adding/removing items while interacting with a NPC could cause an error but it's official behaviour for pre-RE and using items is allowed, too...

**Comment by kyeme** [comment]
Yes please make configuration to disable/enable :D

File: src/map/clif.c (hunk 1 above)
Comment (Nurkz) [reply src/map/clif.c:12838]: You can add there additional check for item usage script state. Which already exists in some scripts (buildin_disableitemuse)

File: conf/map/battle/items.conf [hunk 2]
```diff
@@ -100,10 +100,14 @@ autospell_stacking: false
...
 item_restricted_consumption_type: true
 
-// Enable all NPC to allow changing of equipments while interacting? (Note 1)
-// Script commands 'enable_items/disable_items' will not be override. (see doc/script_commands.txt)
-// Default: true (official)
-item_enabled_npc: true
+// Which item actions are allowed while interacting with NPC? (Note 3)
+// Script commands 'enable_items/disable_items' will not be overridden. (See doc/script_commands.txt.)
+// 0x0 (ITEMENABLEDNPC_NONE)    - Don't allow any item actions.
+// 0x1 (ITEMENABLEDNPC_EQUIP)   - Allow changing equipment.
```
Comment (4144) [inline conf/map/battle/items.conf:106]: probably need add comments about possible exploits for mode 0x1 and 0x2

Comment (4144) [reply conf/map/battle/items.conf:106]: if item usage/equip in npc scripts need rare, may be better add allow use/equip flags into each item?

Comment (Kenpachi2k13) [reply conf/map/battle/items.conf:106]: > probably need add comments about possible exploits for mode 0x1 and 0x2

`0x1` is official in pre-RE as well as in RE and is enabled in Hercules for many years by default.
For `0x2` I actually I can't think of any real exploits and aside from that it's official in pre-RE.

> if item usage/equip in npc scripts need rare, may be better add allow use/equip flags into each item?

It's not rarely needed. In official it's enabled for all NPCs.

**Pull request merged**